agent/setup.go

```diff
@@ -104,7 +104,7 @@ func NewBaseDeps(configLoader ConfigLoader, logOut io.Writer, providedLogger hcl
 
 	var extraSinks []metrics.MetricSink
 	if cfg.IsCloudEnabled() {
-		d.HCP, err = hcp.NewDeps(cfg.Cloud, d.Logger, d.RuntimeConfig.NodeID)
+		d.HCP, err = hcp.NewDeps(cfg.Cloud, d.Logger, cfg.NodeID)
```

Comment:
Made a small change to use `cfg.NodeId` instead. If the NodeID is empty, a new one will be generated and added into the cfg here. It's better to use this than the `RuntimeConfig`